src/sonic-config-engine/minigraph.py

```diff
@@ -1999,6 +2000,28 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
     results['DHCP_SERVER'] = dict((item, {}) for item in dhcp_servers)
     results['DHCP_RELAY'] = dhcp_relay_table
     results['NTP_SERVER'] = dict((item, {}) for item in ntp_servers)
+    # Set default DNS nameserver from dns.j2
+    results['DNS_NAMESERVER'] = {}
+    if os.environ.get("CFGGEN_UNIT_TESTING", "0") == "2":
+        dns_conf = os.path.join(os.path.dirname(__file__), "tests/", "dns.j2")
+    else:
+        dns_conf = "/usr/local/sonic/dns/dns.j2"
+    if os.path.isfile(dns_conf):
+        text = ""
+        with open(dns_conf) as template_file:
+            environment = jinja2.Environment(trim_blocks=True) # nosemgrep
```

Comment:
>nosemgrep [](http://example.com/codeflow?start=65&length=9)

In future PR, please add more code comment to explain what was the semgrep rule triggered, and why it is suppressed.

Reply:
Updated with comment